python/cudf/cudf/tests/test_pandas_interop.py

```diff
@@ -83,6 +83,14 @@ def test_from_pandas_rangeindex():
     assert idx1.name == idx2.name
 
 
+def test_from_pandas_datetimeindex_freq():
```

Comment:
@brandon-b-miller Can we also add `cudf.Index`, `cudf.Index.from_pandas` flows into this pytest? Apart from that, I think this PR is good to go.